### PR TITLE
Best practice for mailing list and github notification

### LINKED ISSUE #1724: ContribEx Guide Needs to Recommend Mailing List Best Practices
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What happened**:
As a Kubernetes community member, it is difficult to attain a good signal-to-noise ratio with regards to GitHub notifications and mailing list messages. 

**What you expected to happen**:
I'd like to get some guidance on how to be an effective communicator in the Kubernetes community. In other words, not be deluged by all the e-mail. Specifically, what to do with GitHub and Gmail to surface the pertinent content and purge the less personally meaningful content.

**How to reproduce it (as minimally and precisely as possible)**:
Join as a Member, help out with more than one SIG.

**Anything else we need to know?**:
N/A

**Environment**:
- Kubernetes version (use `kubectl version`): N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others:

cc @castrojo @bgrant0607 @idvoretskyi 
/sig contributor-experience
/area contributor-guide

### PR DESCRIPTION
**Pull request by nzoueidi**
I would like to have your feedback and inputs for the Mailing list and Github notifications best practice. 
I am thinking that the two parts need to be in the same page for more simplicity.

Once, there is agreement about this file, I will add the links to the top level README and then also to the communication section of the contributor guide.

Fixes #1724

### CODE REVIEW

**Comment by nzoueidi** [comment]
Also, I think the second part of mailing list should have more steps (steps for creating labels and apply them, etc)..

/sig contributor-experience
/area contributor-guide

**Comment by cblecker** [comment]
/ok-to-test
/assign @castrojo 

Assigning Jorge for review.

One thing of note, I'm not sure the root of `communication/` is the right place for this. Also, we typically go with lowercase filenames for these docs, rather than mixed case.

**Comment by nzoueidi** [comment]
> One thing of note, I'm not sure the root of `communication/` is the right place for this. Also, we typically go with lowercase filenames for these docs, rather than mixed case.

I was going to create the file with lowercase, but it will be too long. It would be something like `mailing-list-github-notification-best-practice.md` or it wouldn't be obvious if I keep it the same name but with lowercase `mailinglist-githubnotification-best-practice.md`.

I think though about `mailing-list-github-usage.md`, what do you think?

**Comment by castrojo** [comment]
I would just go with communication/best-practices.md to keep it simple and then let the title speak for itself.  

It will look right when I put it in the right order in the table of contents for the entire contributor guide with the page descriptions. 

Thanks for working on this, let's do the rename and I'll approve and merge, thanks!

**Comment by geekygirldawn** [comment]
This is a great addition! The one thing I'm wondering about is whether the instructions should be specific to Gmail or more general to apply to other email clients. With that said, enough of us use Gmail that I don't really object to having the instructions be specific :)

File: communication/best-practices.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+In order to not get spammed by Github notifications and to get all the 
+notifications in the right place, here is an example of filters to create/apply
+on your Gmail account:
+
+- Stick a blue label on anything kubernetes-related:
+  - Matches: (kubernetes OR k8s OR kubelet OR kube
+  - Do this: Apply label "k8s", Mark it as important
+- Archive my own actions (sending these is an option in github's settings).  
+I send them but archive them, so if I need to see history of an issue I can:
```
Comment (nikhita) [inline communication/best-practices.md]: Maybe rephrase his to remove the `I`s? ;)

File: communication/best-practices.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  - Click on the **Create filter**.
+
+## Examples for setting filters to Kubernetes Github notifications
+
+In order to not get spammed by Github notifications and to get all the 
+notifications in the right place, here is an example of filters to create/apply
+on your Gmail account:
+
+- Stick a blue label on anything kubernetes-related:
+  - Matches: (kubernetes OR k8s OR kubelet OR kube
```
Comment (nikhita) [inline communication/best-practices.md]: If we are looking just for github notifications, this should either be `kubernetes`, `kubernetes-sigs`, etc....

File: communication/best-practices.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  - Do this: Apply label "k8s", Mark it as important
+- Archive my own actions (sending these is an option in github's settings).
```
Comment (nikhita) [inline communication/best-practices.md]: s/github/GitHub

File: communication/best-practices.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+In order to not get spammed by Github notifications and to get all the 
+notifications in the right place, here is an example of filters to create/apply
+on your Gmail account:
+
+- Stick a blue label on anything kubernetes-related:
+  - Matches: (kubernetes OR k8s OR kubelet OR kube
+  - Do this: Apply label "k8s", Mark it as important
+- Archive my own actions (sending these is an option in github's settings).  
+I send them but archive them, so if I need to see history of an issue I can:
+  - Matches: from:noti...@github.com to:(your_a...@noreply.github.com)
```
Comment (nikhita) [inline communication/best-practices.md]: This should be expanded

File: communication/best-practices.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+- Stick a blue label on anything kubernetes-related:
+  - Matches: (kubernetes OR k8s OR kubelet OR kube
+  - Do this: Apply label "k8s", Mark it as important
+- Archive my own actions (sending these is an option in github's settings).  
+I send them but archive them, so if I need to see history of an issue I can:
+  - Matches: from:noti...@github.com to:(your_a...@noreply.github.com) 
+    - Do this: Skip Inbox, Mark as read
+- Skip bot comments:
+  - Matches: (from:(notifi...@github.com) (from:(k8s-merge-robot) OR 
+from:(Kubernetes Bot) OR from:(k8s-ci-robot) OR from:(Submit Queue)))
```
Comment (nikhita) [inline communication/best-practices.md]: There is no Submit Queue anymore :)

Comment (nikhita) [reply communication/best-practices.md]: Also the bot is now called `Kubernetes Prow Robot` and the username would be `k8s-ci-robot`.

File: communication/best-practices.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    - Do this: Skip Inbox, Mark as read
+- Skip push notifications:
+  - Matches: to:(pu...@noreply.github.com) 
+    - Do this: Skip Inbox, Mark as read 
+- Red label on things assigned to me:
```
Comment (nikhita) [inline communication/best-practices.md]: This isn't mentioned below, but `<review_requested@noreply.github.com>` is also a thing and I find it useful to automatically know where review is requested from me... could be useful to just mention it here IMHO :)

File: communication/best-practices.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  - Matches: to:(men...@noreply.github.com) 
+    - Do this: Apply label "gh/p2" 
+- Grey label:
+  - Matches: to:(team_m...@noreply.github.com)
+    - Do this: Apply label "gh/p3"
+  - Matches: to:(aut...@noreply.github.com) 
+    - Do this: Star it, Apply label "gh/me", Mark it as important 
+- Red label:
+  - Matches: (list:"kubernetes.kubernetes.github.com" ("Labelling this PR 
+as size/XL" OR "Labelling this PR as size/XXL")
```
Comment (nikhita) [inline communication/best-practices.md]: The bot does not add the size comments anymore, so this won't work now

File: communication/best-practices.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    - Do this: Apply label "k8s/L"
+- Yellow label:
+  - Matches: list:"kubernetes.kubernetes.github.com" "Labelling this PR as size/M"
+    - Do this: Apply label "k8s/M"
+- Green label:
+  - Matches: (list:"kubernetes.kubernetes.github.com" ("Labelling this PR 
+as size/XS" OR "Labelling this PR as size/S"))
+    - Do this: Apply label "k8s/S"
+- Purple label:
+  - Matches: from:("Kubernetes Submit Queue")
```
Comment (nikhita) [inline communication/best-practices.md]: Submit Queue is :skull: now

File: communication/best-practices.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+- Yellow label:
+  - Matches: to:(men...@noreply.github.com) 
+    - Do this: Apply label "gh/p2" 
+- Grey label:
+  - Matches: to:(team_m...@noreply.github.com)
+    - Do this: Apply label "gh/p3"
+  - Matches: to:(aut...@noreply.github.com) 
+    - Do this: Star it, Apply label "gh/me", Mark it as important 
+- Red label:
+  - Matches: (list:"kubernetes.kubernetes.github.com" ("Labelling this PR
```
Comment (nikhita) [inline communication/best-practices.md]: Maybe add a note that you can also categorize per repo?

For example, I use this:

```
Matches: list:(community.kubernetes.github.com)
Do this: Apply label "k8s/community
```

File: communication/best-practices.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
+# Mailing list and Github usage best practice                                                       
+                                                                                
+The Kubernetes Mailing list or Google Groups functions as the primary means of  
+asynchronous communication for the project's                                    
+[Special Interest Groups (SIG)][sig-list] and [Working Groups (WG)][sig-list].  
+That's why you may want to set your filters in your Gmail account to attain a
```
Comment (nikhita) [inline communication/best-practices.md]: Dawn has a good point! Let's add a note here with something like "If you use Gmail..."

File: communication/best-practices.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+good signal-to-noise ratio with regards to the mailing list messages and Github
+notifications.
+
+- [Example for setting filters to Kubernets Mailing lists](#example-for-setting-filters-to-kubernets-mailing-lists)
+- [Examples for setting filters to Kubernetes Github notifications](#examples-for-setting-filters-to-kubernetes-github-notifications)
+                                                                                
+## Example for setting filters to Kubernets Mailing lists
```
Comment (nikhita) [inline communication/best-practices.md]: IIUC the instructions below in this section show how to create a filter in Gmail, but not specific to Kubernetes mailing lists...

Personally, I use this (if it's helpful):

```
Matches: list:"kubernetes-dev@googlegroups.com"
Do this: Apply label "lists/kubernetes-dev"
```

File: communication/best-practices.md (hunk 1 above)
Comment (idvoretskyi) [reply communication/best-practices.md]: +1, let's make the phrasing more neutral.

**Comment by parispittman** [comment]
Thanks a ton for taking this on @nzoueidi!! 🚀 

Can you link the [mailinglist-guidelines.md](https://github.com/kubernetes/community/blob/master/communication/mailing-list-guidelines.md) here with a line like - "if you are looking to create and manage lists as a chair or other organizer, check this mailing list guideline doc."

**Comment by justaugustus** [comment]
@nzoueidi -- I'd also suggest linking this discussion: https://groups.google.com/d/topic/kubernetes-dev/5qU8irU7_tE/discussion

**Comment by nzoueidi** [comment]
Thank you all for the review. @cblecker @castrojo @geekygirldawn  @nikhita @parispittman Could you PTAL.

File: communication/best-practices.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
+# Mailing list and Github usage best practice                                                       
+                                                                                
+The Kubernetes Mailing list or Google Groups functions as the primary means of  
+asynchronous communication for the project's                                    
+[Special Interest Groups (SIG)][sig-list] and [Working Groups (WG)][sig-list].  
+That's why you may want to set your filters in your email account to attain a
+good signal-to-noise ratio with regards to the mailing list messages and Github
+notifications. All the steps below are basically for Gmail users.
```
Comment (thockin) [inline communication/best-practices.md]: ...but the same ideas should hold for other mail systems

File: communication/best-practices.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+
+Note: If you are looking to create and manage lists as a chair or other organizer, 
+check this mailing list [guideline][mailinglist-guidelines] doc.
+
+- [Example for setting filters to Kubernets Mailing lists](#example-for-setting-filters-to-kubernets-mailing-lists)
+- [Examples for setting filters to Kubernetes Github notifications](#examples-for-setting-filters-to-kubernetes-github-notifications)
+                                                                                
+## Example for setting filters to Kubernets Mailing lists           
+
+It depends on the SIG or/and WG you are involved in, you would like to setup
```
Comment (thockin) [inline communication/best-practices.md]: This sentence is a little awkward, maybe break it to 2 pieces?

File: communication/best-practices.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+  - E.g: Matches: list:"kubernetes-dev@googlegroups.com"
+  - Do this: Apply label "lists/kubernetes-dev"
+
+## Examples for setting filters to Kubernetes Github notifications
+
+In order to not get spammed by Github notifications and to get all the 
+notifications in the right place, here is an example of filters to create/apply
+on your Gmail account:
+
+- Stick a blue label on anything kubernetes-related:
```
Comment (thockin) [inline communication/best-practices.md]: Nothing in this config dictates color, I just happen to use blue :)

File: communication/best-practices.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+  - Matches: (kubernetes OR kubernetes-client OR kubernetes-sigs OR kubernetes-incubator
+OR kubernetes-csi)
+  - Do this: Apply label "k8s", Mark it as important
+- Archive your own actions (sending these is an option in Github's settings).  
+You can send them but also archive them, so whenever you need to see the history of 
+an issue you can:
+  - Matches: from:notifications@github.com to:(your_activity@noreply.github.com)
```
Comment (thockin) [inline communication/best-practices.md]: I think simply "to:your_activity@noreply.github.com" is sufficient

File: communication/best-practices.md [hunk 16]
```diff
@@ -0,0 +1,82 @@
+# Mailing list and Github usage best practice
```
Comment (fejta) [inline communication/best-practices.md]: We should encourage people to use https://gubernator.k8s.io/pr (and [improve it](https://github.com/kubernetes/test-infra/tree/master/gubernator)) rather than everyone (individually no less!) to setup some baroque set of gmail filters.

File: communication/best-practices.md (hunk 12 above)
Comment (nzoueidi) [reply communication/best-practices.md]: yes, but I am not sure that the same steps applies for other mail systems?

File: communication/best-practices.md (hunk 14 above)
Comment (nzoueidi) [reply communication/best-practices.md]: yes, but it is good to show that you can setup filters and categorize them by color.. That's why I kept them

File: communication/best-practices.md (hunk 16 above)
Comment (nzoueidi) [reply communication/best-practices.md]: I think that should be in another page, like having a detailed readme file about gubernator. 
Maybe, we could add a note about gubernator and that we recommend using it?

**Comment by castrojo** [comment]
/lgtm

I have some questions about gubernator's project status, we can do that afterwards during a contribex call. Going to give this ~24h for last call, then I'll approve.

File: communication/best-practices.md [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+good signal-to-noise ratio with regards to the mailing list messages and Github
+notifications. All the steps below are basically for Gmail users.
+
+Note: If you are looking to create and manage lists as a chair or other organizer, 
+check this mailing list [guideline][mailinglist-guidelines] doc.
+
+- [Example for setting filters to Kubernets Mailing lists](#example-for-setting-filters-to-kubernets-mailing-lists)
+- [Examples for setting filters to Kubernetes Github notifications](#examples-for-setting-filters-to-kubernetes-github-notifications)
+                                                                                
+## Example for setting filters to Kubernets Mailing lists
```
Comment (anguslees) [inline communication/best-practices.md]:
```suggestion
## Example for setting filters to Kubernetes Mailing lists           
```

File: communication/best-practices.md [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+check this mailing list [guideline][mailinglist-guidelines] doc.
+
+- [Example for setting filters to Kubernets Mailing lists](#example-for-setting-filters-to-kubernets-mailing-lists)
+- [Examples for setting filters to Kubernetes Github notifications](#examples-for-setting-filters-to-kubernetes-github-notifications)
+                                                                                
+## Example for setting filters to Kubernets Mailing lists           
+
+It depends on the SIG or/and WG you are involved in. 
+You would like to setup your filters in your Gmail account, so you don't need to have 
+all emails structred in one place.
```
Comment (anguslees) [inline communication/best-practices.md]:
```suggestion
all emails structured in one place.
```

File: communication/best-practices.md [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+    - Do this: Star it, Apply label "gh/p1" 
+- Yellow label on things you have been mentioned on:
+  - Matches: to:(mention@noreply.github.com) 
+    - Do this: Apply label "gh/p2" 
+- Grey label:
+  - Matches: to:(team_mention@noreply.github.com)
+    - Do this: Apply label "gh/p3"
+  - Matches: to:(author@noreply.github.com) 
+    - Do this: Star it, Apply label "gh/me", Mark it as important 
+- Skip messages about issues that I am not participating in, but leave them unread:
```
Comment (nikhita) [inline communication/best-practices.md]: The I's should be removed :)

File: communication/best-practices.md (hunk 16 above)
Comment (nikhita) [reply communication/best-practices.md]: +1 about adding a note about gubernator

File: communication/best-practices.md (hunk 19 above)
Comment (nzoueidi) [reply communication/best-practices.md]: Oups, forget that. Thank you Nikhita :)

File: communication/best-practices.md [hunk 20]
```diff
@@ -0,0 +1,82 @@
+# Mailing list and Github usage best practice                                                       
+                                                                                
+The Kubernetes Mailing list or Google Groups functions as the primary means of  
+asynchronous communication for the project's                                    
+[Special Interest Groups (SIG)][sig-list] and [Working Groups (WG)][sig-list].
```
Comment (nikhita) [inline communication/best-practices.md]: Maybe link the SIG and WG sections in the sig list respectively...

File: communication/best-practices.md [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+                                                                                
+The Kubernetes Mailing list or Google Groups functions as the primary means of  
+asynchronous communication for the project's                                    
+[Special Interest Groups (SIG)][sig-list] and [Working Groups (WG)][sig-list].  
+That's why you may want to set your filters in your email account to attain a
+good signal-to-noise ratio with regards to the mailing list messages and Github
+notifications. All the steps below are basically for Gmail users.
+
+Note: If you are looking to create and manage lists as a chair or other organizer, 
+check this mailing list [guideline][mailinglist-guidelines] doc.
```
Comment (nikhita) [inline communication/best-practices.md]: Oh good point about mentioning this! :+1:

File: communication/best-practices.md [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+
+- [Example for setting filters to Kubernets Mailing lists](#example-for-setting-filters-to-kubernets-mailing-lists)
+- [Examples for setting filters to Kubernetes Github notifications](#examples-for-setting-filters-to-kubernetes-github-notifications)
+                                                                                
+## Example for setting filters to Kubernetes Mailing lists           
+
+It depends on the SIG or/and WG you are involved in. 
+You would like to setup your filters in your Gmail account, so you don't need to have 
+all emails structured in one place.
```
Comment (nikhita) [inline communication/best-practices.md]: I'm confused by what "you don't need to have all emails structured in one place" means? :thinking:

File: communication/best-practices.md [hunk 23]
```diff
@@ -0,0 +1,82 @@
...
+
+- In your Gmail account click on **Settings**:
+  - **Filters and Blocked Addresses** -> Scroll down and click **create a new filter**
+  - In the **to** fields write the email of the SIG's Google Group. 
+  - **Create filter** -> Check the box ** Apply the label** and create new
+label by choosing **New label...** in the dropdown list.
+  - Click on the **Create filter**.
+- You can also use filters directly for lists.
+  - E.g: Matches: list:"kubernetes-dev@googlegroups.com"
+  - Do this: Apply label "lists/kubernetes-dev"
```
Comment (nikhita) [inline communication/best-practices.md]: Can we remove the "Eg" from the bullet point and start that point from "Matches"...so that "Matches" and "Do this" will align

File: communication/best-practices.md [hunk 24]
```diff
@@ -0,0 +1,82 @@
...
+  - Matches: to:(your_activity@noreply.github.com) 
+    - Do this: Skip Inbox, Mark as read
+- Skip bot comments:
+  - Matches: (from:(notifications@github.com) (from:(k8s-merge-robot) OR 
+from:(Kubernetes Prow Robot) OR from:(k8s-ci-robot)))
+    - Do this: Skip Inbox, Mark as read
+- Skip push notifications:
+  - Matches: to:(push@noreply.github.com) 
+    - Do this: Skip Inbox, Mark as read 
+- Red label on things assigned to me and/or things request to be reviewed:
```
Comment (nikhita) [inline communication/best-practices.md]: s/me/you

File: communication/best-practices.md [hunk 25]
```diff
@@ -0,0 +1,82 @@
...
+  - Matches: to:(assign@noreply.github.com)
+    - Do this: Star it, Apply label "gh/p0", Mark it as important 
+  - Matches: to:(review_requested@noreply.github.com)
+    - Do this: Star it, Apply label "gh/rq", Mark it as important
```
Comment (nikhita) [inline communication/best-practices.md]: Can we have this as p1 or something? or maybe change the assign, etc labels to `gh/assign`? :thinking:

File: communication/best-practices.md (hunk 22 above)
Comment (nzoueidi) [reply communication/best-practices.md]: maybe, we can say: "you will not have all the incoming emails in one place"?

File: communication/best-practices.md (hunk 25 above)
Comment (nzoueidi) [reply communication/best-practices.md]: I will change just the labels `gh/*`, it looks more good `gh/assign`

**Comment by nzoueidi** [comment]
Last time, it failed due to the pod pending timeout. Triggering another build to make sure that everything is fine. 

/test pull-community-verify

File: communication/best-practices.md (hunk 16 above)
Comment (fejta) [reply communication/best-practices.md]: Any chance we can address this?

**Comment by parispittman** [comment]
@nzoueidi @castrojo can we still include a link to gubernator in the meantime as an additional resource?

File: communication/best-practices.md (hunk 16 above)
Comment (nzoueidi) [reply communication/best-practices.md]: Oups, I thought I added it the note already, sorry - updating it right now.

**Comment by nzoueidi** [comment]
> @nzoueidi @castrojo can we still include a link to gubernator in the meantime as an additional resource?

Thanks @parispittman just added it. PTAL.

File: communication/best-practices.md [hunk 26]
```diff
@@ -0,0 +1,88 @@
...
+    - Do this: Apply label "k8s/community"   
+
+This [thread](https://groups.google.com/forum/#!topic/kubernetes-dev/5qU8irU7_tE/discussion) in kubernetes-dev google group
+is also useful for setting up filters in Gmail.
+ 
+[sig-list]: /sig-list.md#master-sig-list
+[wg-list]: /sig-list.md#master-working-group-list
+[mailinglist-guidelines]: /communication/mailing-list-guidelines.md
+[gubernator]: https://gubernator.k8s.io/pr 
+[repo]: https://github.com/kubernetes/test-infra/tree/master/gubernator
```
[the remark this answers is not in the document]
Comment (nikhita) [reply communication/best-practices.md]: https://git.k8s.io/test-infra/gubernator :)

**Comment by justaugustus** [comment]
I think is looking pretty great. Can we merge and iterate over any remaining feedback?

**Comment by nikhita** [comment]
Retroactive lgtm. Thanks again for getting this in, @nzoueidi!! :100: :1st_place_medal:

**Comment by justaugustus** [comment]
Yayyyyyyyyyyyyyy :rocket:

**Comment by nzoueidi** [comment]
Back to this, I think we should communicate about this in k-dev. What do you think?
